image-loader/app/lib/imaging/FileMetadataReader.scala

```diff
@@ -78,17 +81,50 @@ object FileMetadataReader {
     } getOrElse Map()
 
 
-  def dimensions(image: File): Future[Option[Dimensions]] =
+  def dimensions(image: File, mimeType: Option[String]): Future[Option[Dimensions]] =
     for {
       metadata <- readMetadata(image)
     }
     yield {
-      for {
-        jpegDir <- Option(metadata.getFirstDirectoryOfType(classOf[JpegDirectory]))
+
+      mimeType match {
+
+        case Some("image/jpeg") => for {
+          jpegDir <- Option(metadata.getFirstDirectoryOfType(classOf[JpegDirectory]))
+
+        } yield Dimensions(jpegDir.getImageWidth, jpegDir.getImageHeight)
+
+        case Some("image/png") => for {
+          pngDir <- Option(metadata.getFirstDirectoryOfType(classOf[PngDirectory]))
+
+        } yield {
+          val width = pngDir.getInt(PngDirectory.TAG_IMAGE_HEIGHT)
+          val height = pngDir.getInt(PngDirectory.TAG_IMAGE_WIDTH)
+          Dimensions(width, height)
+        }
+
+        case _ => None
+
       }
-      yield Dimensions(jpegDir.getImageWidth, jpegDir.getImageHeight)
     }
 
+  def exportPngDirectory(metadata: Metadata, mimeType: Option[String]): Map[String, String] = {
```

Comment:
I don't think you should have to hand the mimeType in here. Can you move the case statement up to the calling method?